.github/workflows/ci.yml

```diff
@@ -17,40 +17,60 @@ jobs:
         run: |
           SUSEConnect -p sle-module-basesystem/15.3/x86_64
           zypper install -y unzip-6.00-4.8.13
-
+          zypper install -y jq
+          zypper install -y which
       # Install terraform on self-hosted VM if it doesn't exist
       - name: Install Terraform
         uses: hashicorp/setup-terraform@v1
         with:
           terraform_version: 1.1.4
+          terraform_wrapper: false
+
+  validate-charts:
+    runs-on: self-hosted
+    steps:
+      # Lint Charts
+      run: echo "lint chart to be implemented"
+
+  prepare-environment:
+    needs: [setup-terraform,validate-charts]
+    runs-on: self-hosted
+    steps:
+      - name: Get Base Repo
+        run: |
+          mkdir openstack_env
+          cd openstack_env
+          git clone --recurse-submodules https://gitlab.suse.de/mkravec/scripts.git
+          cd scripts
+
+      - name: Build Validator Container Image
+        run: |
+          cd openstack_env/scripts
+          cp  ~/developer-s390-openrc.sh env.conf
+          ./varke build
 
-  deploy-s390x-vm:
-    needs: setup-terraform
+  run-tests:
+    needs: [prepare-environment]
     runs-on: self-hosted
     steps:
-      # Run terraform script to launch s390x-VM
-      - name: Execute Terraform
+      - name: Run Tests for Longhorn Chart
         run: |
-          source /root/developer-s390-openrc.sh
-          eval $(ssh-agent -s)
-          ssh-add ~/.ssh/id_shared
-          cd openstack_tf_s390x
-          terraform init
-          terraform validate 
-          terraform apply --auto-approve
-      # TODO: Setup RKE2 Cluster
-      # TODO: Run Helm chart linter
-      # TODO: Run Basic tests for specific chart-name
+          cd openstack_env/scripts
+          cp  ~/developer-s390-openrc.sh env.conf
+          chmod 400 data/id_shared
+          sed -i 's/\-it//g' varke 
+          ./varke run -n 1:0 -t longhorn -e SUFFIX=s390x-charts-lh -k
```

Comment:
If you remove `-k` cluster (terraform) will destroy automatically after test runs.
Then you don't have to run destroy-vm step yourself.

Reply:
yeah, I put the -k intentionally so if someone wants to rerun the pipeline and check or cat the logs on this step before destroying the vm, they can do so.